lib/msf/util/payload_cached_size.rb

```diff
@@ -27,6 +27,7 @@ class PayloadCachedSize
       'RC4PASSWORD' => 'Metasploit',
       'DNSZONE' => 'corelan.eu',
       'PEXEC' => '/bin/sh',
+      'HttpUserAgent' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:94.0) Gecko/20100101 Firefox/94.0',
```

Comment:
Why not use one of the Rex::UserAgent strings here, given this is library code?

Reply:
So first I should double-check my understanding of the payload cached size: My understanding is that this is used in certain circumstances to verify that there's enough room in the exploit for the payload, without having to generate it (which is slow). The important thing then for this section of code is to have a representative value to help with that estimation. The HTTP payloads seem to guesstimate this anyway, throwing in some extra room for possible custom headers, proxy options, URL length, etc.: https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/payload/windows/reverse_http.rb#L131-L152

All of that having been said, my reasoning for not using a `Rex::UserAgent` reference here was:

- Using a random UA string breaks unit test expectations around cached payload size being consistent. We could just hard-code one (e.g. use "shortest"), but...
- This doesn't really need to be up-to-date to the latest UA string: it's only used to estimate a payload size, so as long as user agent strings industry-wide don't dramatically change length anytime soon, it should be a reasonable value. Plus:
- Making it refer to `Rex::UserAgent` will likely require us to run the cached payload size updater every time these values are updated (which hopefully we'll do a bit more frequently) to make unit tests (`cached size == actual size`) pass.

So in short: hard-coding it (marginally) reduces the burden of updating the user agent strings.

Reply:
Yep this was a misunderstanding on my side. The way this is done fine makes sense, and thanks for also linking to the additional documentation. After reviewing all of this I'm happy to close this out as resolved.